### PR TITLE
New repo who dis? distribution/reference

### PR DESCRIPTION
**Pull request by dims**
github.com/docker/distribution/reference has a new home github.com/distribution/reference

and a new tag v0.5.0. Let's switch to that.

#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: vendor/github.com/distribution/reference/normalize.go [hunk 1]
```diff
@@ -56,41 +79,53 @@ func ParseNormalizedNamed(s string) (Named, error) {
...
 }
 
-// splitDockerDomain splits a repository name to domain and remotename string.
+// splitDockerDomain splits a repository name to domain and remote-name.
 // If no valid domain is found, the default domain is used. Repository name
 // needs to be already validated before.
 func splitDockerDomain(name string) (domain, remainder string) {
 	i := strings.IndexRune(name, '/')
-	if i == -1 || (!strings.ContainsAny(name[:i], ".:") && name[:i] != "localhost") {
+	if i == -1 || (!strings.ContainsAny(name[:i], ".:") && name[:i] != localhost && strings.ToLower(name[:i]) == name[:i]) {
```
Comment (liggitt) [inline vendor/github.com/distribution/reference/normalize.go:128]: the `strings.ToLower(name[:i]) == name[:i])` bit is interesting and was added in https://github.com/distribution/distribution/pull/2979/files

does this mean an image spec of `FOO/bar` that previously was treated as `docker.io/FOO/bar` is now treated as `FOO/bar`? does dockerhub disallow uppercase in repository names?

Comment (dims) [reply vendor/github.com/distribution/reference/normalize.go:128]: @thaJeztah PTAL

Comment (liggitt) [reply vendor/github.com/distribution/reference/normalize.go:128]: if dockerhub currently requires all namespaces to be lowercase, then this is fine, I just wanted to make sure no one was currently successfully using `Foo/bar` to refer to a `docker.io/Foo/bar` image

Comment (liggitt) [reply vendor/github.com/distribution/reference/normalize.go:128]: ha, we got a unit test failure as well:

```
{Failed;  === RUN   TestParseImageName
    parsers_test.go:52: ParseImageName(ROOT) expects error couldn't parse image name "ROOT": invalid reference format: repository name (library/ROOT) must be lowercase but did not get one
--- FAIL: TestParseImageName (0.00s)
```

looks like we were leaning on the underlying parser to reject this name and it no longer does

Comment (milosgajdos) [reply vendor/github.com/distribution/reference/normalize.go:128]: > if dockerhub currently requires all namespaces to be lowercase,

This is correct, Docker Hub does not allow mixed case, only lowercase, but it foes further than that, `docker` cli doesn't support uppercase repository names either. I am not familiar with other tools like `podman`, etc

Comment (thaJeztah) [reply vendor/github.com/distribution/reference/normalize.go:128]: Thanks Milos! Yes, so it's definitely correct from that perspective; the only _possible_ change is that someone using `HELLO/world` would previously end up on `docker.io/HELLO/world` (and an error likely), and in the new situation, `HOST` would be considered a hostname, so potentially, a request can be made to `http(s)://HELLO/.....`

I don't expect huge issues with that, but it's worth mentioning.

**Comment by dims** [comment]
Gah! missed @liggitt 's comment that the failures were legit. will dig into them.

**Comment by pacoxu** [comment]
Just see this. Should we add the old reference to the unwanted dependencies? See https://github.com/kubernetes/kubernetes/pull/120332/files#diff-09f70f26cff6c34dc0063a45ca43cf5025d653058b06f7966fc8e6f3b9cecd3e.

**Comment by pacoxu** [comment]
/priority important-soon
/triage accepted
LGTM

**Pull request merged**